dialects/sqlite-3-33/src/main/kotlin/app/cash/sqldelight/dialects/sqlite_3_33/grammar/mixins/UpdateStmtLimitedMixin.kt

```diff
@@ -0,0 +1,19 @@
+package app.cash.sqldelight.dialects.sqlite_3_33.grammar.mixins
+
+import app.cash.sqldelight.dialects.sqlite_3_33.grammar.psi.SqliteUpdateStmtLimited
+import com.alecstrong.sql.psi.core.psi.QueryElement
+import com.alecstrong.sql.psi.core.psi.impl.SqlUpdateStmtLimitedImpl
+import com.intellij.lang.ASTNode
+import com.intellij.psi.PsiElement
+
+internal abstract class UpdateStmtLimitedMixin(
+  node: ASTNode,
+) : SqlUpdateStmtLimitedImpl(node) {
+  override fun queryAvailable(child: PsiElement): Collection<QueryElement.QueryResult> {
+    return if (this is SqliteUpdateStmtLimited && updateFrom != null) {
+      updateFrom!!.tablesAvailable(child).map { it.query }
+    } else {
```

Comment:
@AlecStrong this naively works for simple cases, but fails when the subquery names the column using `AS`:

A. Is there a better approach then the one I am taking
B. What is the best way to handle naming a column in the subquery using `AS`?

Reply:
If you swap the rule to just use `join_clause` as I mentioned below you should be able to invoke [queryExposed](https://github.com/AlecStrong/sql-psi/blob/master/core/src/main/kotlin/com/alecstrong/sql/psi/core/psi/mixins/JoinClauseMixin.kt#L92) on it to get the correct resolution

Reply:
If I update the rule to:

```
update_stmt ::= [ {with_clause} ] UPDATE [ OR ROLLBACK | OR ABORT | OR REPLACE | OR FAIL | OR IGNORE ] {qualified_table_name}
  update_set_clause
  [ FROM {join_clause} ]
  [ WHERE <<expr '-1'>> ] {
  implements = "com.alecstrong.sql.psi.core.psi.SqlUpdateStmt"
  mixin = "app.cash.sqldelight.dialects.sqlite_3_33.grammar.mixins.UpdateStmtMixin"
  pin = 4
}
```

how do I get `SqliteUpdateStmt` to have a `getJoinClause` method?

Reply:
you have to get it manually using `findPsiChildOfType<JoinClause>()` or whatever the api is

Reply:
Updated